### PR TITLE
Lint fixes

### PR DESCRIPTION
**Pull request by joshuacolvin0**
Make sure that errors propagated or logged as appropriate
Remove dead code
Fix various other lint suggestions

### CODE REVIEW

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 26
- Added 13
           

Complexity increasing per file
==============================
- packages/arb-util/value/value_test.go  1
- packages/arb-validator/structures/node_test.go  1
- packages/arb-validator-core/ethbridge/globalInbox.go  1
- packages/arb-validator-core/ethbridgetest/value_test.go  1
- packages/arb-evm/evm/log_test.go  1
- packages/arb-validator/nodegraph/stakedNodeGraph.go  1
- packages/arb-validator/challenges/executionChallenger.go  1
- packages/arb-validator/chainobserver/rollupEthBridge_test.go  1
- packages/arb-tx-aggregator/batcher/queue.go  1
- packages/arb-validator/challenges/executionDefender.go  1
- packages/arb-util/hashing/hashing.go  1
         

Complexity decreasing per file
==============================
+ packages/arb-tx-aggregator/batcher/batcher.go  -1
+ packages/arb-validator/challenges/testHelper_test.go  -2
+ packages/arb-validator/challenges/defender.go  -3
         

Clones added
============
- packages/arb-validator/chainlistener/validatorChainListener.go  5
         

Clones removed
==============
+ packages/arb-validator/challenges/testHelper_test.go  -2
+ packages/arb-validator/challenges/executionDefender.go  -1
+ packages/arb-validator/challenges/inboxTopDefender.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=6359992&bid=20643800)

File: packages/arb-validator/structures/node.go [hunk 1]
```diff
@@ -134,8 +130,8 @@ func NewRandomInvalidNodeFromValidPrev(
 		common.NewTimeBlocks(common.RandBigInt()),
 	)
 
-	_ = nextNode.UpdateInvalidOpinion()
-	return nextNode
+	err := nextNode.UpdateInvalidOpinion()
```
Comment (hkalodner) [inline packages/arb-validator/structures/node.go:133]: I think we should be able to safely ignore this error. Instead it'd be good to add the `kind != valprotocol.ValidChildType` check into this function and panic if it's violated. This is only used in tests so it's nice to keep the API simple

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: packages/arb-validator/structures/node.go (hunk 1 above)
Comment (joshuacolvin0) [reply packages/arb-validator/structures/node.go:133]: done

File: packages/arb-validator/nodegraph/stakedNodeGraph.go [hunk 2]
```diff
@@ -365,30 +370,7 @@ func (sng *StakedNodeGraph) CheckChallengeOpportunityAny(staker *Staker) *Challe
...
-
-func (sng *StakedNodeGraph) makeConfirmOpp(
-	nodeOps []valprotocol.ConfirmNodeOpportunity,
-	confNodes []*structures.Node,
-	currentTime common.TimeTicks,
-	stakerAddrs []common.Address,
-) (*valprotocol.ConfirmOpportunity, []*structures.Node) {
+func (sng *StakedNodeGraph) makeConfirmOpp(nodeOps []valprotocol.ConfirmNodeOpportunity, confNodes []*structures.Node, currentTime common.TimeTicks, stakerAddrs []common.Address) (*valprotocol.ConfirmOpportunity, []*structures.Node, error) {
 	nodeLimit := len(nodeOps)
 	for nodeLimit > 0 {
```
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-validator/nodegraph/stakedNodeGraph.go:375]: refactored

File: packages/arb-evm/evm/log.go [hunk 3]
```diff
@@ -148,16 +150,15 @@ func NewLogFromValue(val value.Value) (Log, error) {
...
 	data := []value.Value{
 		value.NewValueFromAddress(l.Address),
 		inbox.BytesToByteStack(l.Data),
 	}
 	for _, topic := range l.Topics {
 		data = append(data, value.NewIntValue(new(big.Int).SetBytes(topic.Bytes())))
 	}
-	val, _ := value.NewTupleFromSlice(data)
-	return val
+	return value.NewTupleFromSlice(data)
```
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-evm/evm/log.go:161]: This one actually varies with the number of topics.  Ethereum sets topic limit to four, but best to leave test in place to catch corruption, etc.

File: packages/arb-evm/message/transfer.go [hunk 4]
```diff
@@ -30,7 +30,8 @@ type Eth struct {
 
 func NewEthFromData(data []byte) Eth {
 	destAddress, data := extractAddress(data)
-	payment, data := extractUInt256(data)
+	// Last value returned is array of bytes
```
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-evm/message/transfer.go:33]: Changed all related comments to `// Last value returned is not an error type`

File: packages/arb-validator/chainlistener/validatorChainListener.go [hunk 5]
```diff
@@ -257,12 +263,15 @@ func (lis *ValidatorChainListener) StakeMoved(
...
+		_, err := InitiateChallenge(ctx, lis.actor, opp)
+		if err != nil {
+			log.Printf("Stake %v unable to initiate challenge: %v", ev.Staker, err)
+		}
 	}
 }
 
 func (lis *ValidatorChainListener) challengeStakerIfPossible(
-	ctx context.Context,
+	_ context.Context,
```
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-validator/chainlistener/validatorChainListener.go:274]: done

File: packages/arb-validator/nodegraph/stakedNodeGraph.go [hunk 6]
```diff
@@ -409,8 +391,9 @@ func (sng *StakedNodeGraph) makeConfirmOpp(
 				CurrentLatestConfirmed: sng.latestConfirmed.Hash(),
 				StakerAddresses:        stakerAddrs,
 				StakerProofs:           proofs,
-			}, confNodes
+			}, confNodes, nil
 		}
 	}
-	panic("Unreachable code")
+
+	return nil, nil, fmt.Errorf("unreachable code")
```
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-validator/nodegraph/stakedNodeGraph.go:398]: refactored and commented

File: packages/arb-validator/nodegraph/stakedNodeGraph.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-validator/nodegraph/stakedNodeGraph.go:375]: refactored

File: packages/arb-validator/structures/node.go (hunk 1 above)
Comment (joshuacolvin0) [reply packages/arb-validator/structures/node.go:133]: added appropriate panic

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 22
- Added 13
           

Complexity increasing per file
==============================
- packages/arb-util/value/value_test.go  1
- packages/arb-evm/evm/log_test.go  1
- packages/arb-validator/challenges/executionDefender.go  1
- packages/arb-validator/structures/node_test.go  1
- packages/arb-validator/challenges/executionChallenger.go  1
- packages/arb-tx-aggregator/batcher/queue.go  1
- packages/arb-validator-core/ethbridge/globalInbox.go  1
- packages/arb-util/hashing/hashing.go  1
- packages/arb-validator-core/ethbridgetest/value_test.go  1
         

Complexity decreasing per file
==============================
+ packages/arb-validator/challenges/defender.go  -3
+ packages/arb-tx-aggregator/batcher/batcher.go  -1
+ packages/arb-validator/nodegraph/stakedNodeGraph.go  -1
+ packages/arb-validator/challenges/testHelper_test.go  -2
         

Clones added
============
- packages/arb-validator/chainlistener/validatorChainListener.go  5
         

Clones removed
==============
+ packages/arb-validator/challenges/inboxTopDefender.go  -1
+ packages/arb-validator/challenges/executionDefender.go  -1
+ packages/arb-validator/challenges/testHelper_test.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=6359992&bid=20643800)

**Review by hkalodner — DISMISSED** [review]
LGTM. Approved pending circleci completion

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 22
- Added 13
           

Complexity increasing per file
==============================
- packages/arb-validator/structures/node_test.go  1
- packages/arb-validator-core/ethbridgetest/value_test.go  1
- packages/arb-util/hashing/hashing.go  1
- packages/arb-tx-aggregator/batcher/queue.go  1
- packages/arb-validator/challenges/executionDefender.go  1
- packages/arb-evm/evm/log_test.go  1
- packages/arb-validator-core/ethbridge/globalInbox.go  1
- packages/arb-validator/challenges/executionChallenger.go  1
- packages/arb-util/value/value_test.go  1
         

Complexity decreasing per file
==============================
+ packages/arb-validator/nodegraph/stakedNodeGraph.go  -1
+ packages/arb-validator/challenges/testHelper_test.go  -2
+ packages/arb-tx-aggregator/batcher/batcher.go  -1
+ packages/arb-validator/challenges/defender.go  -3
         

Clones added
============
- packages/arb-validator/chainlistener/validatorChainListener.go  5
         

Clones removed
==============
+ packages/arb-validator/challenges/inboxTopDefender.go  -1
+ packages/arb-validator/challenges/executionDefender.go  -1
+ packages/arb-validator/challenges/testHelper_test.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=6359992&bid=20643800)

**Comment by hkalodner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 22
- Added 12
           

Complexity increasing per file
==============================
- packages/arb-evm/evm/log_test.go  1
- packages/arb-tx-aggregator/batcher/queue.go  1
- packages/arb-validator/structures/node_test.go  1
- packages/arb-validator/challenges/executionDefender.go  1
- packages/arb-util/value/value_test.go  1
- packages/arb-validator/challenges/executionChallenger.go  1
- packages/arb-validator-core/ethbridgetest/value_test.go  1
- packages/arb-util/hashing/hashing.go  1
- packages/arb-validator-core/ethbridge/globalInbox.go  1
         

Complexity decreasing per file
==============================
+ packages/arb-validator/challenges/testHelper_test.go  -2
+ packages/arb-validator/challenges/defender.go  -3
+ packages/arb-tx-aggregator/batcher/batcher.go  -1
         

Clones added
============
- packages/arb-validator/chainlistener/validatorChainListener.go  5
         

Clones removed
==============
+ packages/arb-validator/challenges/testHelper_test.go  -2
+ packages/arb-validator/challenges/executionDefender.go  -1
+ packages/arb-validator/challenges/inboxTopDefender.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=6359992&bid=20643800)

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**